presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneJoinColumns.java

```diff
@@ -49,38 +47,31 @@ public Pattern getPattern()
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator, Session session)
     {
-        ProjectNode parent = (ProjectNode) node;
+        return pushDownProjectOff(
```

Comment:
IMO it is difficult to tell what this function does. Maybe if you would use method reference instead of lambda then it would be better.